python/mxnet/io.py

```diff
@@ -617,6 +620,111 @@ def getpad(self):
         return self.pad
 
 
+class SFrameImageIter(SFrameIter):
+    """Image Data Iterator from SFrame
+    Provide the SFrameIter like interface with options to normalize and augment image data.
+
+    Parameters
+    ----------
+    sframe : SFrame object
+        source SFrame
+    data_field : string
+        image data field of the SFrame.
+    label_field : string, optional
+        label field in SFrame
+    batch_size : int, optional
+        batch size
+    mean_r : float, optional
+        normalize the image by subtracting the mean value of r channel, or the first channel for
+    mean_g : float, optional
+        normalize the image by subtracting the mean value of g channel
+    mean_b : float, optional
+        normalize the image by subtracting the mean value of b channel
+    mean_nd : np.ndarray, optional
+        normalize the image by subtracting the ndarray of mean pixel values.
+        The mean_nd array stores the pixel values in the order of [height, width, channel]
+        This option will surpress mean_r, mean_g, and mean_b.
+    scale : float, optional
+        multiply each pixel value by the scale (this operation is performed after mean subtraction)
+    **kwargs :
+        placeholder for new parameters
+
+    Examples
+    --------
+    >>> import sframe as sf
+    >>> import mxnet as mx
+
+    >>> image_data = sf.SFrame('http://s3.amazonaws.com/dato-datasets/mnist/sframe/train')
+    >>> image_data_iter = mx.io.SFrameImageIter(sframe=data, data_field=['image'], label_field='label', batch_size=100,
+                                                mean_r=117, scale=0.5)
+
+    Notes
+    -----
+    - Image column must contain images of the same size.
+    """
+
+    def __init__(self, sframe, data_field, label_field=None, batch_size=1,
+                 data_name='data', label_name='softmax_label',
+                 mean_r=0.0,
+                 mean_g=0.0,
+                 mean_b=0.0,
+                 mean_nd=None,
+                 scale=1.0,
+                 **kwargs):
+        super(SFrameImageIter, self).__init__(sframe, data_field, label_field, batch_size,
+                                              data_name, label_name)
+
+        # Mean subtraction parameters
+        self._rgb_mask = np.zeros(self.data_shape)
+        if mean_nd is None:
+            nchannels = self.data_shape[1]
+            mean_per_channel = [mean_r, mean_g, mean_b][:nchannels]
+            for i in range(nchannels):
+                self._rgb_mask[:, i, :, :] = mean_per_channel[i]
+        elif type(mean_nd) == np.ndarray:
```

Comment:
We don't need to swap axes, and if pass with mean_nd, I think we don't need to do anything on it because it is generated by record io